docs/conf.py

```diff
@@ -84,6 +86,148 @@ def git_describe_version(original_version):
 version = git_describe_version(tvm.__version__)
 release = version
 
+
+def monkey_patch(module_name, func_name):
```

Comment:
you might consider using `unittest.mock.patch` here since it's in the stdlib

Reply:
Using stdlib functions is always good, but `unittest.mock.patch` does not quite fit this use case. I can't use `@patch` as a decorator, as the decorator must be used on the function **calling** the function we wish to monkey-patch (e.g. `save_rst_example`).

I could just call `patch` as a function afterwards, but `patch` also does not include functionality for wrapping functions. It would be possible to make `patch()` work, but I think writing our own decorator is cleaner.

I've clarified the comment to better explain this.

Reply:
ahh i see. potentially you could theoretically override sphinx-gallery's `reset_modules` and use the [optionally-passed `when` parameter](https://github.com/sphinx-gallery/sphinx-gallery/blob/4484d35bb556bfe081156f210904abce8732ffbf/sphinx_gallery/scrapers.py#L592) to decide whether to monkey-patch or un-monkey-patch. i agree it's sort of equivalent both ways, i just wish there was a way to avoid "poisoning" sphinx-gallery permanently by importing conf.py.

Reply:
I agree that `reset_modules` is a valid approach, but I think going that route would make the changes less readable. Permanently altering `sphinx-gallery` is not desirable, but it may be the cleanest option other than forking `sphinx-gallery`.